packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -307,8 +307,9 @@ function getValidTelemetryProps(obj: any, keysToOmit: Set<string>): ITelemetryPr
         const val = obj[key];
         // ensure only valid props get logged, since props of logging error could be in any shape
         const validProp = filterValidTelemetryProps(val);
-        if (isTaggedTelemetryPropertyValue(val)) {
-            props[key] = val;
+        const validTaggedProp = filterValidTelemetryProps(val.value);
+        if (isTaggedTelemetryPropertyValue(val) && validTaggedProp !== null) {
+            props[key] = validTaggedProp;
```

Comment:
This looks great in terms of functionality (with exception of using `null` as special value), but expensive in terms of perf as we do extra (unneeded) check all the time. Better would be something like that:
```ts
function filterValidTelemetryProps(x: any): TelemetryEventPropertyType {
    ...
    console.error(`UnSupported Format of Logging Error Property for key ${key}:`, val);
    return "REDACTED (arbitrary object);
}

...

if (isTaggedTelemetryPropertyValue(val)) {
     props[key] = filterValidTelemetryProps(val.value);
} else {
     props[key] = filterValidTelemetryProps(val);
}
```

The only (minor) downside of this flow - we keep property tagged even if it has no interesting data anymore.
I think it's fine - restructuring code to avoid it will be a bit messy.

While null is not allowed type in TelemetryEventPropertyType, this code should not assume that - if we ever change it to support null, it better work correctly and not require changes.